### PR TITLE
fix: Optimize write time cost when periodic full compaction

### PR DESCRIPTION
**Pull request by MachaelLee**
## Which issue does this PR close?

Closes #

## Rationale for this change
Currently cost time of writing data will increase during periodic full compaction, resulting in write requests failing.

## What changes are included in this PR?
* make aliyun params `keep_alive` configurable
* use `io_runtime` to execute object store, not `compact_runtime`
* use `max_ongoing_task` to reduce traffic peak of compaction tasks
* remove compaction after flush

### CODE REVIEW

File: analytic_engine/src/instance/flush_compaction.rs [hunk 1]
```diff
@@ -293,47 +292,24 @@ impl Flusher {
         };
         let flush_job = async move { flush_task.run().await };
 
-        // TODO: The immediate compaction after flush is not a good idea because it may
-        // block on the write procedure.
-        if let Some(compaction_scheduler) = opts.compact_after_flush.clone() {
```
Comment (chunshao90) [inline analytic_engine/src/instance/flush_compaction.rs:298]: Remove `compact_after_flush`.

File: analytic_engine/src/instance/flush_compaction.rs [hunk 2]
```diff
@@ -356,8 +332,11 @@ impl FlushTask {
             .set_last_flush_time(time::current_time_millis());
 
         info!(
-            "Instance flush memtables done, table:{}, table_id:{}, request_id:{}",
-            self.table_data.name, self.table_data.id, request_id
+            "Instance flush memtables done, table:{}, table_id:{}, request_id:{}, cost:{}",
```
Comment (chunshao90) [inline analytic_engine/src/instance/flush_compaction.rs:335]:
```suggestion
            "Instance flush memtables done, table:{}, table_id:{}, request_id:{}, cost_ms:{}",
```

File: analytic_engine/src/instance/flush_compaction.rs [hunk 3]
```diff
@@ -293,47 +292,24 @@ impl Flusher {
...
-                    opts,
-                    &self.runtime,
-                    &table_data.metrics,
-                )
-                .await
-        }
+        flush_scheduler
+            .flush_sequentially(
+                flush_job,
+                async {},
```
Comment (Rachelint) [inline analytic_engine/src/instance/flush_compaction.rs:298]: Remove this param? Unused now.

File: analytic_engine/src/compaction/scheduler.rs [hunk 4]
```diff
@@ -78,6 +79,7 @@ impl Default for SchedulerConfig {
             // flush_interval default is 5h.
             max_unflushed_duration: ReadableDuration(Duration::from_secs(60 * 60 * 5)),
             memory_limit: ReadableSize::gb(4),
+            max_pending_compaction_tasks: MAX_PENDING_COMPACTION_TASKS,
```
Comment (Rachelint) [inline analytic_engine/src/compaction/scheduler.rs:82]: I think `const` is unnecessary here.

File: components/object_store/src/metrics.rs [hunk 5]
```diff
@@ -129,19 +187,45 @@ impl ObjectStore for StoreWithMetrics {
...
+            .runtime
+            .spawn(async move { store.head(&loc).await })
+            .await
+            .map_err(|source| StoreError::Generic {
+                store: METRICS,
+                source: Box::new(source),
+            })?;
+
+        info!(
+            "Object store with metrics head cost:{}, location:{}",
```
Comment (chunshao90) [inline components/object_store/src/metrics.rs:211]:
```suggestion
            "Object store with metrics head cost_ms:{}, location:{}",
```

File: components/object_store/src/metrics.rs [hunk 6]
```diff
@@ -112,12 +141,41 @@ impl ObjectStore for StoreWithMetrics {
...
+        let result = self
+            .runtime
+            .spawn(async move { store.get_range(&loc, range.clone()).await })
+            .await
+            .map_err(|source| StoreError::Generic {
+                store: METRICS,
+                source: Box::new(source),
+            })??;
+        info!(
+            "Object store with metrics get_range cost:{}, location:{}, thread:{}-{:?}",
```
Comment (chunshao90) [inline components/object_store/src/metrics.rs:172]:
```suggestion
            "Object store with metrics get_range cost_ms:{}, location:{}, thread:{}-{:?}",
```

File: analytic_engine/src/compaction/scheduler.rs (hunk 4 above)
Comment (MachaelLee) [reply analytic_engine/src/compaction/scheduler.rs:82]: well, I will remove MAX_PENDING_COMPACTION_TASKS and MAX_ONGOING_TASKS.

File: analytic_engine/src/instance/flush_compaction.rs (hunk 2 above)
Comment (MachaelLee) [reply analytic_engine/src/instance/flush_compaction.rs:335]: I guess it is better to do like this:
```suggestion
            "Instance flush memtables done, table:{}, table_id:{}, request_id:{}, cost:{}ms",
```

File: analytic_engine/src/instance/flush_compaction.rs (hunk 3 above)
Comment (MachaelLee) [reply analytic_engine/src/instance/flush_compaction.rs:291]: Which param? I can't get it.

File: analytic_engine/src/setup.rs [hunk 7]
```diff
@@ -408,6 +412,7 @@ impl ObjectStorePicker for OpenedStorages {
 // ```
 fn open_storage(
     opts: StorageOptions,
+    runtime: Arc<Runtime>,
```
Comment (chunshao90) [inline analytic_engine/src/setup.rs:415]: Write a comment stating that the runtime needs to be a separate io runtime.

File: components/object_store/src/metrics.rs [hunk 8]
```diff
@@ -100,7 +110,29 @@ impl ObjectStore for StoreWithMetrics {
...
+        let res = self
+            .runtime
+            .spawn(async move { store.put_multipart(&loc).await })
+            .await
+            .map_err(|source| StoreError::Generic {
+                store: METRICS,
+                source: Box::new(source),
+            })?;
+
+        info!(
```
Comment (chunshao90) [inline components/object_store/src/metrics.rs:128]: Use trace level.

File: components/object_store/src/metrics.rs [hunk 9]
```diff
@@ -112,12 +144,41 @@ impl ObjectStore for StoreWithMetrics {
...
+        let loc = location.clone();
+        let result = self
+            .runtime
+            .spawn(async move { store.get_range(&loc, range.clone()).await })
+            .await
+            .map_err(|source| StoreError::Generic {
+                store: METRICS,
+                source: Box::new(source),
+            })??;
+        info!(
```
Comment (chunshao90) [inline components/object_store/src/metrics.rs:174]: Use trace level.

File: components/object_store/src/metrics.rs [hunk 10]
```diff
@@ -129,19 +190,45 @@ impl ObjectStore for StoreWithMetrics {
...
+        let response = self
+            .runtime
+            .spawn(async move { store.head(&loc).await })
+            .await
+            .map_err(|source| StoreError::Generic {
+                store: METRICS,
+                source: Box::new(source),
+            })?;
+
+        info!(
```
Comment (chunshao90) [inline components/object_store/src/metrics.rs:213]: Use trace level.

File: components/object_store/src/metrics.rs (hunk 8 above)
Comment (MachaelLee) [reply components/object_store/src/metrics.rs:128]: I have change level to 'debug'

File: analytic_engine/src/compaction/scheduler.rs [hunk 11]
```diff
@@ -626,8 +626,11 @@ impl ScheduleWorker {
 
             // This will spawn a background job to purge ssts and avoid schedule thread
```
Comment (jiacai2050) [inline analytic_engine/src/compaction/scheduler.rs:627]: Remove this comments

File: components/object_store/src/metrics.rs [hunk 12]
```diff
@@ -100,7 +110,29 @@ impl ObjectStore for StoreWithMetrics {
...
+            .map_err(|source| StoreError::Generic {
+                store: METRICS,
+                source: Box::new(source),
+            })?;
+
+        trace!(
+            "Object store with metrics put_multipart cost:{}ms, location:{}, thread:{}-{:?}",
+            instant.elapsed().as_millis(),
+            location,
+            thread_name,
```
Comment (jiacai2050) [inline components/object_store/src/metrics.rs:132]: Move L114 here, so it won't get executed in default log level.

File: analytic_engine/src/compaction/scheduler.rs (hunk 11 above)
Comment (MachaelLee) [reply analytic_engine/src/compaction/scheduler.rs:627]: I update this comment.

**Review by jiacai2050 — APPROVED** [review]
LGTM

**Pull request merged**